tests/Custom/TimeoutTest.php

```diff
@@ -76,6 +79,32 @@ public function testShortPageLoadTimeoutThrowsException(): void
 
         $this->expectException(DriverException::class);
         $this->expectExceptionMessage('Page failed to load: ');
+
         $session->visit($this->pathTo('/page_load.php?sleep=2'));
     }
+
+    /**
+     * @group legacy
+     * @dataProvider deprecatedPageLoadDataProvider
+     */
+    public function testDeprecatedShortPageLoadTimeoutThrowsException(string $type): void
+    {
+        $session = $this->getSession();
+        $driver = $session->getDriver();
+        \assert($driver instanceof WebdriverClassicDriver);
+
+        $driver->setTimeouts([$type => 500]);
+
+        $this->expectDeprecation('Using "' . $type . '" timeout type is deprecated, please use "page" instead');
```

Comment:
1. Have you checked that absense of deprecation notice causes a test to fail on both PHPUni 9.x and PHPUnit 10.x?
2. Do we test on PHPUnit 10.x at all?

Reply:
1. Of course...what would be the point of "expectDeprecation" otherwise? 😄 
2. Apparently not...  dependency is at `^9.6.8`. We should probably upgrade since that version is [on life support](https://phpunit.de/supported-versions.html).

Why do you ask?

Note that this is using the symfony functionality, not the original PHPUnit one (by using a trait).

Reply:
Side note, apparently there's a lot of failures now... could it be related to a recent driver-testsuite update?
https://github.com/minkphp/webdriver-classic-driver/actions/runs/11650940031

It doesn't happen to me locally, even after updating composer dependencies. 🤔

Reply:
I reran CI for the main branch and it finished successfully (also rerun it for this branch several times and it always failed).

That would only mean that something in this branch is breaking things badly later on.

Reply:
I did a lot of changes, but somehow I cannot get GH workflow to work reliably.

At this point, it would be great to get #43 merged first to "hide" some of the (duplicate) changes in this PR (after rebasing). That would have the added benefit of decreasing the surface area of the problem (i.e. that mink/session is not involved).